crates/adapters/src/transport/datagen.rs

```diff
@@ -31,6 +31,122 @@ use rand_distr::{Distribution, Zipf};
 use serde_json::{to_writer, Map, Value};
 use tokio::sync::Notify;
 
+fn range_as_i64(field: &str, range: &Option<(Value, Value)>) -> AnyResult<Option<(i64, i64)>> {
+    match range {
+        None => Ok(None),
+        Some((Value::Number(a), Value::Number(b))) => {
+            let a = a
+                .as_i64()
+                .ok_or_else(|| anyhow!("Invalid min range for field {:?}", field))?;
+            let b = b
+                .as_i64()
+                .ok_or_else(|| anyhow!("Invalid max range for field {:?}", field))?;
+            Ok(Some((a, b)))
+        }
+        _ => Err(anyhow!(
+            "Range values must be integers for field {:?}",
+            field
+        )),
+    }
+}
+
+/// Tries to parse a range as a date range which is days since UNIX epoch
+/// but we can specify either as a number or a string.
+fn parse_range_for_date(
+    field: &str,
+    range: &Option<(Value, Value)>,
+) -> AnyResult<Option<(i64, i64)>> {
+    match range {
+        None => Ok(None),
+        Some((Value::Number(a), Value::Number(b))) => {
+            let a = a
+                .as_i64()
+                .ok_or_else(|| anyhow!("Invalid min range for field {:?}", field))?;
+            let b = b
+                .as_i64()
+                .ok_or_else(|| anyhow!("Invalid max range for field {:?}", field))?;
+            Ok(Some((a, b)))
+        }
+        Some((Value::String(a), Value::String(b))) => {
+            let unix_date: NaiveDate = NaiveDate::from_ymd_opt(1970, 1, 1).unwrap();
```

Comment:
If it's malformed will you get a decent message?

Reply:
line 74 and 76 will do the error handlign for malformed user strings

Reply:
what is the user experience?

Reply:
they'll get an error from the adapter -- currently they'll have to inspect the manager log but once we land error handling in manager this will be there too